src/components/logic/operate.js

```diff
@@ -0,0 +1,116 @@
+  switch (operator) {
+    case '+':
+      return operand1 + operand2;
+    case '-':
+      return operand1 - operand2;
+    case 'x':
+      return operand1 * operand2;
+    case '/':
+      return operand1 / operand2;
+
+    default:
+      return 0;
+  }
+};
+
+export const numClickHandler = (value, calc, setCalc) => {
+  let newScreenValue = calc.screenValue;
+  if (newScreenValue === '0') {
+    newScreenValue = '';
+  }
+  newScreenValue += value.toString();
+  const num = parseFloat(newScreenValue);
+
+  setCalc({
+    ...calc,
+    operand2: num,
+    screenValue: newScreenValue,
+  });
+};
+
+export const signClickHandler = (operator, calc, setCalc) => {
+  if (calc.operator) {
+    const result = operate(calc.operator, parseFloat(calc.operand1), parseFloat(calc.operand2));
+    setCalc({
+      ...calc,
+      operand1: result,
+      operand2: null,
+      result,
+      operator,
+      screenValue: result.toString(),
+    });
+  } else {
+    setCalc({
+      ...calc,
+      operator,
+      operand1: parseFloat(calc.screenValue),
+      screenValue: '',
+    });
+  }
+};
+
+export const equalClickHandler = (calc, setCalc) => {
+  let { result } = calc;
+  if (calc.operator && calc.operand2 !== null) {
+    result = operate(calc.operator, parseFloat(calc.operand1), parseFloat(calc.operand2));
+  }
+  setCalc({
+    ...calc,
+    operator: '',
+    operand1: 0,
+    operand2: 0,
+    result,
+    screenValue: result.toString(),
+  });
+};
+
+export const resetClickHandler = (calc, setCalc) => {
+  setCalc({
+    screenValue: '0',
+    result: null,
+    operand1: null,
+    operand2: null,
+    operator: null,
+  });
+};
+
+export const percentClickHandler = (value, calc, setCalc) => {
+  const num = parseFloat(calc.screenValue);
+  const percent = operate('/', num, 100);
+  const result = operate(calc.operator, parseFloat(calc.operand1), percent);
+  setCalc({
+    ...calc,
+    operand2: percent,
+    screenValue: percent.toString(),
+    result,
+  });
+};
+
+export const commaClickHandler = (value, calc, setCalc) => {
+  const num = parseFloat(calc.screenValue);
+
+  if (typeof num === 'number') {
+    setCalc({
+      ...calc,
+      screenValue: `${num.toLocaleString('en-US', { maximumFractionDigits: 10 })}.`,
+      operand2: calc.operator ? num : parseFloat(calc.operand2),
+      operand1: calc.operator ? parseFloat(calc.operand1) : parseFloat(calc.screenValue),
+      result: calc.operator ? operate(calc.operator, parseFloat(calc.operand1), num) : calc.result,
+    });
+  }
+};
+
+export const invertClickHandler = (value, calc, setCalc) => {
+  const num = parseFloat(calc.screenValue);
+
+  const newNum = -num;
+
+  setCalc({
+    ...calc,
+    screenValue: newNum.toString(),
+    operand1: calc.operator ? parseFloat(calc.operand1) : newNum,
+    operand2: calc.operator ? newNum : parseFloat(calc.operand2),
+    result: calc.operator ? operate(calc.operator, parseFloat(calc.operand1), newNum) : calc.result,
+  });
+};
```

Comment:
- [ ] [REQUIRED] To make work easy and concentrate on how to work with React events, you are advised to use the already developed math logic needed to make the Calculator component functional. Kindly ensure that you use this file as recommended [operate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js). Please copy the file into your logic directory then analyze the file and be sure you understand what they do.

- Kindly refer to the project requirement:
<img width="924" alt="Screenshot 2023-04-27 at 17 22 51" src="https://user-images.githubusercontent.com/58692590/234927595-43010649-6ac8-4daf-a927-e829a5eb8177.png">